lib/NMISNG/Node.pm

```diff
@@ -2648,12 +2648,10 @@ sub collect_node_data
 	elsif ($howdiditgo->{skipped}) {}
 	else
 	{
-		$self->nmisng->log->error("($catchall_data->{name}), Ignoring entry: collect_node_data encountered error $anyerror");
-		$self->handle_down( sys => $S, type => "snmp", details => $howdiditgo->{snmp_error} )
-			if ( $howdiditgo->{snmp_error} );
-		$self->handle_down( sys => $S, type => "wmi", details => $howdiditgo->{wmi_error} )
-				if ( $howdiditgo->{wmi_error} );
-		next;  # OMK-10384 - Node.pm simply returns if any OID is not found. It should simply generate an error message and skip to the next OID.
+		my $error_result = $self->handle_sys_get_data_error( sys => $S, caller => "collect_node_data", 
+			section => $section, index => "catchall", catchall_data => $catchall_data);
+		# return if there is no snmp session
+		return 0 if ( $error_result == 10 );
```

Comment:
Do we want to use 'enums' for this in the future, any other error conditions we would want to catch and give a human response for?

Reply:
We do, I think we should make a plan before throwing them in.